homework2/hw1.py

```diff
@@ -0,0 +1,133 @@
+"""
+Given a file containing text.
+Complete using only default collections:
+    1) Find 10 longest words consisting from largest
+    amount of unique symbols
+    2) Find rarest symbol for document
+    3) Count every punctuation char
+    4) Count every non ascii char
+    5) Find most common non ascii char for document
+"""
+
+import re
+import sys
+from typing import List
+from unicodedata import category
+
+
+def get_longest_diverse_words(file_path: str) -> List[str]:
+    """
+    Returns 10 longest words consisting from largest
+    amount of unique symbols
+
+    :param file_path: File to process
+    :return: Sorted list of longest words
+    """
+
+    words = set()
+    with open(file_path, encoding="unicode_escape") as fi:
+        for line in fi:
+            words.update(re.findall(r"\w+", line.lower()))
+    long_ten = sorted(
+        words, key=lambda w: (len(w), len(set(w))), reverse=True
+    )[:10]
+    return sorted(long_ten)
+
+
+def get_rarest_char(file_path: str) -> str:
+    """
+    Returns rarest non punctuation symbol for document
+
+    :param file_path: File to process
+    :return: Rarest symbol
+    """
+
+    # get unicode punctuation set
+    punc_u = (chr(i) for i in range(sys.maxunicode + 1))
+    punctuation = set(c for c in punc_u if category(c).startswith("P"))
+
+    counter = dict()
```

Comment:
its ok to init new dictionaries with dict(), but counter = {} method considered as more simple and preferred
some linter test will fail because of it